### ISSUE TITLE
ZMQ_HEARTBEAT_TIMEOUT option is not work

### ISSUE THREAD

**Issue by youyoujushi** [zeromq/libzmq#3188]
My zeromq version is 4.2.5, and my code compile success. I set ZMQ_HEARTBEAT_IVL with 2 seconds, set ZMQ_HEARTBEAT_TIMEOUT with 5 seconds. I send request in a loop,the server response one request per second(call sleep(1) ) , but only after send two request, the send operate is blocked.If I set ZMQ_HEARTBEAT_IVL with 3 seconds, the send operate is blocked after send 3 request. It seems ZMQ_HEARTBEAT_IVL influence the timeout. And every send request,I received a response. I don't know why ,can you help me?

/**********client code*****************/
```
#include <zmq.hpp>
#include
#include
int main ()
{
// Prepare our context and socket
zmq::context_t context (1);
zmq::socket_t socket (context, ZMQ_REQ);

int num = 2000;
size_t len = 4;
socket.setsockopt(ZMQ_HEARTBEAT_IVL,&num,len);

num = 5000;
socket.setsockopt(ZMQ_HEARTBEAT_TIMEOUT,&num,len);

std::cout << "Connecting to hello world server..." << std::endl;
socket.connect ("tcp://localhost:5555");

//  Do 10 requests, waiting each time for a response
for (int request_nbr = 0; request_nbr != 10; request_nbr++) {
    zmq::message_t request (5);
    memcpy (request.data (), "Hello", 5);
    std::cout << "Sending Hello " << request_nbr << "..." << std::endl;
    socket.send (request);

    //  Get the reply.
    zmq::message_t reply;
    socket.recv (&reply);
    std::cout << "Received World " << request_nbr << std::endl;
}
return 0;

}
```

/*******server code********/
```
#include <zmq.hpp>
#include <string>
#include <iostream>
#ifndef _WIN32
#include <unistd.h>
#else
#include <windows.h>

#define sleep(n)	Sleep(n)
#endif

int main () {
    //  Prepare our context and socket
    zmq::context_t context (1);
    zmq::socket_t socket (context, ZMQ_REP);
    socket.bind ("tcp://*:5555"); 

    while (true) {
        zmq::message_t request;

        //  Wait for next request from client
        socket.recv (&request);
        std::cout << "Received Hello" << std::endl;

        //  Do some 'work'
    	sleep(1);

        //  Send reply back to client
        zmq::message_t reply (5);
        memcpy (reply.data (), "World", 5);
        socket.send (reply);
    }
    return 0;
}
```

**Comment by bluca** [comment]
Duplicate of https://github.com/zeromq/libzmq/issues/3060 already fixed in master
